diff --git a/src/gallium/auxiliary/gallivm/lp_bld_debug.cpp b/src/gallium/auxiliary/gallivm/lp_bld_debug.cpp
index 86617d4..a8903cc 100644
--- a/src/gallium/auxiliary/gallivm/lp_bld_debug.cpp
+++ b/src/gallium/auxiliary/gallivm/lp_bld_debug.cpp
@@ -212,7 +212,9 @@ disassemble(const void* func, llvm::raw_ostream & Out)
    std::string Error;
    const Target *T = TargetRegistry::lookupTarget(Triple, Error);
 
-#if HAVE_LLVM >= 0x0300
+#if HAVE_LLVM >= 0x0304
+   OwningPtr<const MCAsmInfo> AsmInfo(T->createMCAsmInfo(*T->createMCRegInfo(Triple), Triple));
+#elif HAVE_LLVM >= 0x0300
    OwningPtr<const MCAsmInfo> AsmInfo(T->createMCAsmInfo(Triple));
 #else
    OwningPtr<const MCAsmInfo> AsmInfo(T->createAsmInfo(Triple));
@@ -226,9 +228,9 @@ disassemble(const void* func, llvm::raw_ostream & Out)
 #if HAVE_LLVM >= 0x0300
    const MCSubtargetInfo *STI = T->createMCSubtargetInfo(Triple, sys::getHostCPUName(), "");
    OwningPtr<const MCDisassembler> DisAsm(T->createMCDisassembler(*STI));
-#else 
+#else
    OwningPtr<const MCDisassembler> DisAsm(T->createMCDisassembler());
-#endif 
+#endif
    if (!DisAsm) {
       Out << "error: no disassembler for target " << Triple << "\n";
       return 0;
