diff --git a/xwayland/dnd.c b/xwayland/dnd.c
index daeb08d..f0f2457 100644
--- a/xwayland/dnd.c
+++ b/xwayland/dnd.c
@@ -37,7 +37,6 @@
 
 #include "cairo-util.h"
 #include "compositor.h"
-#include "xserver-server-protocol.h"
 #include "hash.h"
 
 static void
diff --git a/xwayland/launcher.c b/xwayland/launcher.c
index f37cbe4..f0a87f5 100644
--- a/xwayland/launcher.c
+++ b/xwayland/launcher.c
@@ -33,7 +33,6 @@
 #include <signal.h>
 
 #include "xwayland.h"
-#include "xserver-server-protocol.h"
 
 
 static int
diff --git a/xwayland/window-manager.c b/xwayland/window-manager.c
index c8810a9..b8b80b5 100644
--- a/xwayland/window-manager.c
+++ b/xwayland/window-manager.c
@@ -38,7 +38,6 @@
 
 #include "cairo-util.h"
 #include "compositor.h"
-#include "xserver-server-protocol.h"
 #include "hash.h"
 
 struct wm_size_hints {
@@ -550,7 +549,7 @@ weston_wm_window_send_configure_notify(struct weston_wm_window *window)
 	configure_notify.override_redirect = 0;
 	configure_notify.pad1 = 0;
 
-	xcb_send_event(wm->conn, 0, window->id, 
+	xcb_send_event(wm->conn, 0, window->id,
 		       XCB_EVENT_MASK_STRUCTURE_NOTIFY,
 		       (char *) &configure_notify);
 }
@@ -558,7 +557,7 @@ weston_wm_window_send_configure_notify(struct weston_wm_window *window)
 static void
 weston_wm_handle_configure_request(struct weston_wm *wm, xcb_generic_event_t *event)
 {
-	xcb_configure_request_event_t *configure_request = 
+	xcb_configure_request_event_t *configure_request =
 		(xcb_configure_request_event_t *) event;
 	struct weston_wm_window *window;
 	uint32_t mask, values[16];
@@ -626,7 +625,7 @@ our_resource(struct weston_wm *wm, uint32_t id)
 static void
 weston_wm_handle_configure_notify(struct weston_wm *wm, xcb_generic_event_t *event)
 {
-	xcb_configure_notify_event_t *configure_notify = 
+	xcb_configure_notify_event_t *configure_notify =
 		(xcb_configure_notify_event_t *) event;
 	struct weston_wm_window *window;
 
@@ -683,7 +682,7 @@ weston_wm_create_surface(struct wl_listener *listener, void *data)
 		    wl_resource_get_id(surface->resource)) {
 			xserver_map_shell_surface(window, surface);
 			break;
-		}	
+		}
 }
 
 static void
@@ -707,7 +706,7 @@ weston_wm_window_activate(struct wl_listener *listener, void *data)
 		client_message.data.data32[0] = wm->atom.wm_take_focus;
 		client_message.data.data32[1] = XCB_TIME_CURRENT_TIME;
 
-		xcb_send_event(wm->conn, 0, window->id, 
+		xcb_send_event(wm->conn, 0, window->id,
 			       XCB_EVENT_MASK_SUBSTRUCTURE_REDIRECT,
 			       (char *) &client_message);
 
@@ -983,7 +982,7 @@ weston_wm_window_draw_decoration(void *data)
 			/* We leave an extra pixel around the X window area to
 			 * make sure we don't sample from the undefined alpha
 			 * channel when filtering. */
-			pixman_region32_init_rect(&window->surface->pending.opaque, 
+			pixman_region32_init_rect(&window->surface->pending.opaque,
 						  x - 1, y - 1,
 						  window->width + 2,
 						  window->height + 2);
@@ -2256,7 +2255,7 @@ xserver_map_shell_surface(struct weston_wm_window *window,
 	if (!shell_interface->get_primary_view)
 		return;
 
-	window->shsurf = 
+	window->shsurf =
 		shell_interface->create_shell_surface(shell_interface->shell,
 						      window->surface,
 						      &shell_client);
