Index: lib/Makefile
===================================================================
--- lib/Makefile
+++ lib/Makefile
@@ -96,7 +96,9 @@
 	lldbPluginPlatformFreeBSD.a \
 	lldbPluginPlatformPOSIX.a \
 	lldbPluginPlatformKalimba.a \
-	lldbPluginPlatformAndroid.a
+	lldbPluginPlatformAndroid.a \
+	lldbPluginScriptInterpreterNone.a \
+	lldbPluginScriptInterpreterPython.a
 
 # Because GCC requires RTTI enabled for lldbCore (see source/Core/Makefile) it is
 # necessary to also link the clang rewriter libraries so vtable references can
Index: source/Plugins/Makefile
===================================================================
--- source/Plugins/Makefile
+++ source/Plugins/Makefile
@@ -32,7 +32,8 @@
 	SystemRuntime/MacOSX \
 	SymbolVendor/ELF \
 	MemoryHistory/asan \
-	InstrumentationRuntime/AddressSanitizer
+	InstrumentationRuntime/AddressSanitizer \
+	ScriptInterpreter/Python ScriptInterpreter/None
 
 ifeq ($(HOST_OS),Darwin)
 PARALLEL_DIRS += Process/MacOSX-Kernel
Index: source/Plugins/ScriptInterpreter/None/Makefile
===================================================================
--- /dev/null
+++ source/Plugins/ScriptInterpreter/None/Makefile
@@ -0,0 +1,14 @@
+##===- source/Plugins/ScriptInterpreter/None/Makefile ------------*- Makefile -*-===##
+# 
+#                     The LLVM Compiler Infrastructure
+#
+# This file is distributed under the University of Illinois Open Source
+# License. See LICENSE.TXT for details.
+# 
+##===----------------------------------------------------------------------===##
+
+LLDB_LEVEL := ../../../..
+LIBRARYNAME := lldbPluginScriptInterpreterNone
+BUILD_ARCHIVE = 1
+
+include $(LLDB_LEVEL)/Makefile
Index: source/Plugins/ScriptInterpreter/Python/Makefile
===================================================================
--- /dev/null
+++ source/Plugins/ScriptInterpreter/Python/Makefile
@@ -0,0 +1,14 @@
+##===- source/Plugins/ScriptInterpreter/Python/Makefile ------------*- Makefile -*-===##
+# 
+#                     The LLVM Compiler Infrastructure
+#
+# This file is distributed under the University of Illinois Open Source
+# License. See LICENSE.TXT for details.
+# 
+##===----------------------------------------------------------------------===##
+
+LLDB_LEVEL := ../../../..
+LIBRARYNAME := lldbPluginScriptInterpreterPython
+BUILD_ARCHIVE = 1
+
+include $(LLDB_LEVEL)/Makefile
