diff --git a/bindings/ocaml/backends/Makefile b/bindings/ocaml/backends/Makefile
index 7698d6e..07fc760 100644
--- a/bindings/ocaml/backends/Makefile
+++ b/bindings/ocaml/backends/Makefile
@@ -20,6 +20,7 @@ ExtraMakefiles = $(PROJ_OBJ_DIR)/Makefile.common
 
 include $(LEVEL)/Makefile.config
 include $(LEVEL)/Makefile.common
+include ../Makefile.ocaml
 
 all-local:: all-backends
 clean-local:: clean-backends
