diff --git a/src/nouveau_glamor.h b/src/nouveau_glamor.h
index fb6565d..0897aaa 100644
--- a/src/nouveau_glamor.h
+++ b/src/nouveau_glamor.h
@@ -11,6 +11,16 @@
 #define GLAMOR_FOR_XORG 1
 #include "glamor.h"
 
+#ifndef GLAMOR_INVERTED_Y_AXIS
+#define GLAMOR_INVERTED_Y_AXIS 0
+#endif
+#ifndef GLAMOR_USE_SCREEN
+#define GLAMOR_USE_SCREEN 0
+#endif
+#ifndef GLAMOR_USE_PICTURE_SCREEN
+#define GLAMOR_USE_PICTURE_SCREEN 0
+#endif
+
 Bool nouveau_glamor_pre_init(ScrnInfoPtr scrn);
 Bool nouveau_glamor_init(ScreenPtr screen);
 Bool nouveau_glamor_create_screen_resources(ScreenPtr screen);
