diff --git a/radeon/radeon_surface.c b/radeon/radeon_surface.c
index 9c3a192..8a1fe7d 100644
--- a/radeon/radeon_surface.c
+++ b/radeon/radeon_surface.c
@@ -1914,7 +1914,7 @@ static void cik_get_2d_params(struct radeon_surface_manager *surf_man,
         sample_split = 1;
         break;
     case CIK__SAMPLE_SPLIT__2:
-        sample_split = 1;
+        sample_split = 2;
         break;
     case CIK__SAMPLE_SPLIT__4:
         sample_split = 4;
