diff --git a/fontproto.h b/fontproto.h
index 1ab8f37..097769e 100644
--- a/fontproto.h
+++ b/fontproto.h
@@ -25,28 +25,28 @@ authorization from The XFree86 Project Inc..
 
 /* Externally provided functions required by libXfont */
 
-extern int RegisterFPEFunctions ( NameCheckFunc name_func, 
-				  InitFpeFunc init_func, 
-				  FreeFpeFunc free_func, 
-				  ResetFpeFunc reset_func, 
-				  OpenFontFunc open_func, 
-				  CloseFontFunc close_func, 
-				  ListFontsFunc list_func, 
-				  StartLfwiFunc start_lfwi_func, 
-				  NextLfwiFunc next_lfwi_func, 
-				  WakeupFpeFunc wakeup_func, 
-				  ClientDiedFunc client_died, 
-				  LoadGlyphsFunc load_glyphs, 
-				  StartLaFunc start_list_alias_func, 
-				  NextLaFunc next_list_alias_func, 
+extern int RegisterFPEFunctions ( NameCheckFunc name_func,
+				  InitFpeFunc init_func,
+				  FreeFpeFunc free_func,
+				  ResetFpeFunc reset_func,
+				  OpenFontFunc open_func,
+				  CloseFontFunc close_func,
+				  ListFontsFunc list_func,
+				  StartLfwiFunc start_lfwi_func,
+				  NextLfwiFunc next_lfwi_func,
+				  WakeupFpeFunc wakeup_func,
+				  ClientDiedFunc client_died,
+				  LoadGlyphsFunc load_glyphs,
+				  StartLaFunc start_list_alias_func,
+				  NextLaFunc next_list_alias_func,
 				  SetPathFunc set_path_func);
 
 extern int GetDefaultPointSize ( void );
 
-extern int init_fs_handlers ( FontPathElementPtr fpe, 
+extern int init_fs_handlers ( FontPathElementPtr fpe,
 			      BlockHandlerProcPtr block_handler);
-extern void remove_fs_handlers ( FontPathElementPtr fpe, 
-				 BlockHandlerProcPtr block_handler, 
+extern void remove_fs_handlers ( FontPathElementPtr fpe,
+				 BlockHandlerProcPtr block_handler,
 				 Bool all );
 
 extern int client_auth_generation ( ClientPtr client );
@@ -74,7 +74,7 @@ extern FontPtr  CreateFontRec (void);
 extern void  DestroyFontRec (FontPtr font);
 extern Bool     _FontSetNewPrivate (FontPtr        /* pFont */,
 				    int            /* n */,
-				    pointer        /* ptr */);
+				    void *         /* ptr */);
 extern int      AllocateFontPrivateIndex (void);
 extern void ResetFontPrivateIndex (void);
 
diff --git a/fontstruct.h b/fontstruct.h
index 97e771a..c62ebd6 100644
--- a/fontstruct.h
+++ b/fontstruct.h
@@ -122,14 +122,14 @@ typedef struct _Font {
     void        (*unload_font) (FontPtr         /* font */);
     void        (*unload_glyphs) (FontPtr         /* font */);
     FontPathElementPtr fpe;
-    pointer     svrPrivate;
-    pointer     fontPrivate;
-    pointer     fpePrivate;
+    void        *svrPrivate;
+    void        *fontPrivate;
+    void        *fpePrivate;
     int		maxPrivate;
-    pointer	*devPrivates;
+    void        **devPrivates;
 }           FontRec;
 
-#define FontGetPrivate(pFont,n) ((n) > (pFont)->maxPrivate ? (pointer) 0 : \
+#define FontGetPrivate(pFont,n) ((n) > (pFont)->maxPrivate ? (void *) 0 : \
 			     (pFont)->devPrivates[n])
 
 #define FontSetPrivate(pFont,n,ptr) ((n) > (pFont)->maxPrivate ? \
@@ -149,14 +149,14 @@ typedef struct _FontPathElement {
     char       *name;
     int         type;
     int         refcount;
-    pointer     private;
+    void       *private;
 }           FontPathElementRec;
 
 typedef Bool (*NameCheckFunc) (char *name);
 typedef int (*InitFpeFunc) (FontPathElementPtr fpe);
 typedef int (*FreeFpeFunc) (FontPathElementPtr fpe);
 typedef int (*ResetFpeFunc) (FontPathElementPtr fpe);
-typedef int (*OpenFontFunc) ( pointer client,
+typedef int (*OpenFontFunc) ( void *client,
 			      FontPathElementPtr fpe,
 			      Mask flags,
 			      char* name,
@@ -168,55 +168,55 @@ typedef int (*OpenFontFunc) ( pointer client,
 			      char** aliasName,
 			      FontPtr non_cachable_font);
 typedef void (*CloseFontFunc) (FontPathElementPtr fpe, FontPtr pFont);
-typedef int (*ListFontsFunc) (pointer client,
+typedef int (*ListFontsFunc) (void *client,
 			      FontPathElementPtr fpe,
 			      char* pat,
 			      int len,
 			      int max,
 			      FontNamesPtr names);
 
-typedef int (*StartLfwiFunc) (pointer client,
+typedef int (*StartLfwiFunc) (void *client,
 			      FontPathElementPtr fpe,
 			      char* pat,
 			      int len,
 			      int max,
-			      pointer* privatep);
+			      void ** privatep);
 
-typedef int (*NextLfwiFunc) (pointer client,
+typedef int (*NextLfwiFunc) (void *client,
 			     FontPathElementPtr fpe,
 			     char** name,
 			     int* namelen,
 			     FontInfoPtr* info,
 			     int* numFonts,
-			     pointer private);
+			     void *private);
 
 typedef int (*WakeupFpeFunc) (FontPathElementPtr fpe,
 			      unsigned long* LastSelectMask);
 
-typedef void (*ClientDiedFunc) (pointer client,
+typedef void (*ClientDiedFunc) (void *client,
 			       FontPathElementPtr fpe);
 
-typedef int (*LoadGlyphsFunc) (pointer client,
+typedef int (*LoadGlyphsFunc) (void *client,
 			       FontPtr pfont,
 			       Bool range_flag,
 			       unsigned int nchars,
 			       int item_size,
 			       unsigned char* data);
 
-typedef int (*StartLaFunc) (pointer client,
+typedef int (*StartLaFunc) (void *client,
 			    FontPathElementPtr fpe,
 			    char* pat,
 			    int len,
 			    int max,
-			    pointer* privatep);
+			    void ** privatep);
 
-typedef int (*NextLaFunc) (pointer client,
+typedef int (*NextLaFunc) (void *client,
 			   FontPathElementPtr fpe,
 			   char** namep,
 			   int* namelenp,
 			   char** resolvedp,
 			   int* resolvedlenp,
-			   pointer private);
+			   void *private);
 
 typedef void (*SetPathFunc)(void);
 
